src/Tizen.Security.TEEC/Tizen.Security.TEEC/Libteec.cs

```diff
@@ -340,14 +344,37 @@ public sealed class Session
             for (int i=0; i < 4; ++i) shm[i] = null;
         }
 
+        // internal since class is sealed
+        internal void Dispose(bool disposing)
+        {
+            if (disposed) {
+                return ;
+            }
+            if (opened) {
+                Interop.Libteec.CloseSession(session_imp);
+                opened = false;
+            }
+            Marshal.FreeHGlobal(this.opptr);
+            Marshal.FreeHGlobal(this.session_imp);
+            this.opptr = IntPtr.Zero;
+            this.session_imp = IntPtr.Zero;
+            disposed = true;
+        }
+
         /// <summary>
         /// Destructor of the class.
         /// </summary>
         ~Session()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Disposable interface implememtation.
+        /// </summary>
+        public void Dispose()
         {
             Close();
```

Comment:
Please add 
`GC.SuppressFinalize(this);`
It make disable the finalizer

Reply:
Close have to be identical to Dispose
* providing method Close(), in addition to the Dispose(), if close is standard terminology in the area.  it is important that you make the Close implementation identical to Dispose *

To make so, I put a call to Close() from Dispose(), and implementation of Close() calls Dispose(true); and GC.SuppressFinalize(this);
So I don;t understand your point.

Reply:
This object has two state.
 1. Created : Resource allocated.
 2. Opened : Session was established

OpenXX method transit to Opened state from Created

In my opinion, Close is an API that corresponds to OpenXX

So, If Open was transit to Opened from Created
Close should transit to Created from Opened

It is more resonable interface and more flexible to use.

Reply:
It's my point
http://dotnetguts.blogspot.kr/2007/06/difference-between-close-and-dispose.html?m=1